### PR TITLE
Suppress warning in clang about exit time destructor

### PR DESCRIPTION
**Pull request by BenjaminW3**
Fix exit-time-destructor warning mentioned in #673.

### CODE REVIEW

**Comment by ax3l** [comment]
hm... but the warning is actually true. That's not a fix, but a disabling of the diagnostics.

**Comment by BenjaminW3** [comment]
Yes, the warning is true but we explicitly wanted this to happen in that way. I remember the PR and the discussion with @psychocoderHPC where he introduced the change.

File: include/alpaka/atomic/AtomicStdLibLock.hpp [hunk 1]
```diff
@@ -23,6 +23,8 @@
 
 #include <alpaka/atomic/Traits.hpp>
 
+#include <alpaka/core/BoostPredef.h>
```
Comment (tdd11235813) [inline include/alpaka/atomic/AtomicStdLibLock.hpp]: suffix to hpp: `<alpaka/core/BoostPredef.hpp>`

Comment (BenjaminW3) [reply include/alpaka/atomic/AtomicStdLibLock.hpp]: done

**Pull request merged**